lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -531,59 +743,17 @@ bool SessionStateEnforcer::static_rule_init() {
   rule2.set_id("rule2");
   rule2.set_priority(10);
   FlowDescription fd2;
-  fd2.mutable_match()->set_ipv4_src("192.168.0.0/16");
   fd2.mutable_match()->set_direction(FlowMatch_Direction_DOWNLINK);
   fd2.set_action(FlowDescription_Action_PERMIT);
   rule2.mutable_flow_list()->Add()->CopyFrom(fd2);
   reqpdr2.mutable_activate_flow_req()->mutable_dynamic_rules()->Add()->CopyFrom(
       rule2);
-  GlobalRuleList.insert_rule(2, reqpdr2);
-
-  // subscriber Id 1 to PDR 1 and FAR 1
-  pdr_map_.insert(std::pair<std::string, uint32_t>("imsi00000000001", 1));
-  pdr_map_.insert(std::pair<std::string, uint32_t>("imsi00000000001", 2));
-  pdr_map_.insert(std::pair<std::string, uint32_t>("IMSI001222333", 1));
-  pdr_map_.insert(std::pair<std::string, uint32_t>("IMSI001222333", 2));
-  reqpdr1.set_pdr_id(++count);
-  reqpdr1.mutable_set_gr_far()->set_far_id(count);
-  reqpdr1.mutable_pdi()->set_ue_ip_adr("192.168.128.12");
-  GlobalRuleList.insert_rule(3, reqpdr1);
-  reqpdr2.set_pdr_id(++count);
-  reqpdr2.mutable_set_gr_far()->set_far_id(count);
-  reqpdr2.mutable_pdi()->set_ue_ip_adr("192.168.128.12");
-  reqpdr2.mutable_set_gr_far()
-      ->mutable_fwd_parm()
-      ->mutable_outr_head_cr()
-      ->set_o_teid(200 + count);
-  GlobalRuleList.insert_rule(4, reqpdr2);
-  pdr_map_.insert(std::pair<std::string, uint32_t>("imsi00000000002", 3));
-  pdr_map_.insert(std::pair<std::string, uint32_t>("imsi00000000002", 4));
-  reqpdr1.set_pdr_id(++count);
-  reqpdr1.mutable_set_gr_far()->set_far_id(count);
-  reqpdr1.mutable_pdi()->set_ue_ip_adr("192.168.128.13");
-  GlobalRuleList.insert_rule(5, reqpdr1);
-  reqpdr2.set_pdr_id(++count);
-  reqpdr2.mutable_set_gr_far()->set_far_id(count);
-  reqpdr2.mutable_pdi()->set_ue_ip_adr("192.168.128.13");
-  reqpdr2.mutable_set_gr_far()
-      ->mutable_fwd_parm()
-      ->mutable_outr_head_cr()
-      ->set_o_teid(200 + count);
-  GlobalRuleList.insert_rule(6, reqpdr2);
-  pdr_map_.insert(std::pair<std::string, uint32_t>("imsi00000000003", 5));
-  pdr_map_.insert(std::pair<std::string, uint32_t>("imsi00000000003", 6));
-  reqpdr1.set_pdr_id(++count);
-  reqpdr1.mutable_set_gr_far()->set_far_id(count);
-  reqpdr1.mutable_pdi()->set_ue_ip_adr("192.168.128.14");
-  GlobalRuleList.insert_rule(7, reqpdr1);
-  reqpdr2.set_pdr_id(++count);
-  reqpdr2.mutable_set_gr_far()->set_far_id(count);
-  reqpdr2.mutable_pdi()->set_ue_ip_adr("192.168.128.14");
-  reqpdr2.mutable_set_gr_far()
-      ->mutable_fwd_parm()
-      ->mutable_outr_head_cr()
-      ->set_o_teid(200 + count);
-  GlobalRuleList.insert_rule(8, reqpdr2);
+  GlobalRuleList.insert_rule(DEFAULT_DOWN_LINK_PDR_ID, reqpdr2);
   return true;
 }
+
+void SessionStateEnforcer::set_upf_node(std::string node_id, std::string addr) {
+  pipelined_client_->set_upf_node(node_id, addr);

```

Comment:
don't store anything in an API client... Tuck in the UPF IP address into session context...

Reply:
"set_upf_node" is called when we receive these info from UPF as part of initial Node message. Now the qn -- where we do store this info ? There is no other place, other than "Pipleline client, for now 4g/5g client is same "PipelinedClient", In future we shud separate or inherit to create 4g pipelined object and 5g pipelined object separately. Even separating that we have issue while sending , so for now we are using one common Pipeliend client object.

Reply:
Removed from Api Client